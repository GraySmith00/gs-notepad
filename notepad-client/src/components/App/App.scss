.app {
  display: flex;
  flex-direction: column;
  align-items: center;

  height: 100vh;

  background: #4ecdc4; /* fallback for old browsers */
  background: -webkit-linear-gradient(
    to right,
    #556270,
    #4ecdc4
  ); /* Chrome 10-25, Safari 5.1-6 */
  background: linear-gradient(
    to right,
    #556270,
    #4ecdc4
  ); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */

  h1 {
    margin: 5rem 0;
    font-family: 'Satisfy', cursive;
    font-size: 8rem;
    color: #fff;
  }

  &__inner {
    background-color: rgba(#f3f5fc, 0.9);
    width: 60%;
    height: 60vh;
    border-radius: 10px;
    box-shadow: 1rem 1rem 4rem rgba(0, 0, 0, 0.5);
  }

  &__inner-top {
    background-color: #bcc0ce;
    width: 100%;
    height: 35%;
    padding: 3rem;
    border-top-left-radius: 10px;
    border-top-right-radius: 10px;
  }

  &__filters {
    display: flex;
    justify-content: space-between;
    width: 30%;
  }

  &__inner-bottom {
    overflow: scroll;
    height: 65%;
    padding: 3rem;
  }
}

@media only screen and (max-width: 1000px) {
  .app__inner {
    width: 80%;
  }
}
