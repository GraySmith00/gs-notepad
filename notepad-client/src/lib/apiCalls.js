export const postNote = async note => {
  const url = process.env.REACT_APP_POST_NOTE;
  const response = await fetch(url, {
    method: 'POST',
    body: JSON.stringify(note),
    headers: {
      'Content-Type': 'application/json'
    }
  });

  if (response.status > 299) {
    const error = await response.json();
    throw new Error(error.message);
  }

  const data = await response.json();
  return data;
};

export const getNotes = async () => {
  const url = process.env.REACT_APP_GET_NOTES;
  const response = await fetch(url);
  const data = await response.json();
  return data;
};
