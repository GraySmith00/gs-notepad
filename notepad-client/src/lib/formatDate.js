import moment from 'moment';

export const formatDate = timeStamp => {
  return moment(timeStamp).format('L');
};
