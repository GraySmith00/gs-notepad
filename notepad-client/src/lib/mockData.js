export const mockNotes = [
  {
    createdAt: 1545951483012,
    text: 'buy groceries',
    tag: 'hobby',
    id: 'dd5aa440-0a2a-11e9-93fa-a7949507e226'
  },
  {
    createdAt: 1545951433152,
    text: 'do laundry',
    tag: 'work',
    id: 'bfa29c00-0a2a-11e9-93fa-a7949507e226'
  },
  {
    createdAt: 1545951460092,
    text: 'play golf',
    tag: 'hobby',
    id: 'cfb153c0-0a2a-11e9-93fa-a7949507e226'
  },
  {
    createdAt: 1545951504338,
    text: 'buy gifts',
    tag: 'personal',
    id: 'ea10bb20-0a2a-11e9-93fa-a7949507e226'
  },
  {
    createdAt: 1545451632475,
    text: 'set up tv',
    tag: 'personal',
    id: '425ca4b0-0a2b-11e9-93fa-a7949507e226'
  },
  {
    createdAt: 1545951440518,
    text: 'run errands',
    tag: 'work',
    id: 'c4069260-0a2a-11e9-93fa-a7949507e226'
  }
];
