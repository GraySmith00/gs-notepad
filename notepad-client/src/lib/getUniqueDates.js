import { formatDate } from './formatDate';

export const getUniqueDates = notes => {
  const dates = notes.reduce((dates, note) => {
    const formattedDate = formatDate(note.createdAt);

    if (!dates[formattedDate]) {
      dates[formattedDate] = 1;
    }

    return dates;
  }, {});

  return Object.keys(dates);
};
