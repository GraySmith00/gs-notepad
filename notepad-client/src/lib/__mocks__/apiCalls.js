import { mockNotes } from '../mockData';

export const getNotes = jest.fn().mockImplementation(() => {
  return Promise.resolve(mockNotes);
});

export const postNote = jest.fn().mockImplementation(() => {
  return Promise.resolve(mockNotes[0]);
});
