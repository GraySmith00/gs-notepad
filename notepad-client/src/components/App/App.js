import React, { Component } from 'react';
import './App.css';

import { postNote, getNotes } from '../../lib/apiCalls';

import NoteForm from '../NoteForm';
import NotesList from '../NotesList';
import TagFilter from '../TagFilter';
import DateFilter from '../DateFilter';

class App extends Component {
  state = {
    notes: [],
    tagFilter: null,
    dateFilter: null,
    getError: null,
    postError: null
  };

  async componentDidMount() {
    try {
      const notes = await getNotes();
      this.setState({ notes, getError: null });
    } catch (error) {
      this.setState({ getError: error.message });
    }
  }

  addNote = async note => {
    try {
      const savedNote = await postNote(note);
      this.setState({
        notes: [savedNote, ...this.state.notes],
        postError: null
      });
    } catch (error) {
      this.setState({ postError: error.message });
    }
  };

  setTagFilter = value => {
    if (value === 'all') value = null;
    this.setState({ tagFilter: value });
  };

  setDateFilter = value => {
    if (value === 'all') value = null;
    this.setState({ dateFilter: value });
  };

  render() {
    const { notes, tagFilter, dateFilter, postError } = this.state;

    return (
      <div className="App">
        <h1>NotePad</h1>
        <NoteForm addNote={this.addNote} postError={postError} />
        <DateFilter setDateFilter={this.setDateFilter} notes={notes} />
        <TagFilter setTagFilter={this.setTagFilter} />
        <NotesList
          notes={notes}
          tagFilter={tagFilter}
          dateFilter={dateFilter}
        />
      </div>
    );
  }
}

export default App;
