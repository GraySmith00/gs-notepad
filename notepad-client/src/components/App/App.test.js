import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import { postNote } from '../../lib/apiCalls';

jest.mock('../../lib/apiCalls');

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call getNotes', async () => {
    const note = {
      text: 'buy groceries',
      tag: 'hobby'
    };

    await wrapper.instance().addNote(note);

    expect(postNote).toHaveBeenCalled();
  });

  it('should add a note to state', async () => {
    const note = {
      text: 'buy groceries',
      tag: 'hobby'
    };

    const currentNotesLength = wrapper.state().notes.length;

    await wrapper.instance().addNote(note);

    expect(wrapper.state().notes.length).toEqual(currentNotesLength + 1);
  });

  it('should set a tag filter', () => {
    wrapper.instance().setTagFilter('work');
    expect(wrapper.state().tagFilter).toEqual('work');
  });

  it('should set the tag filter to null if the value is all', () => {
    wrapper.instance().setTagFilter('all');
    expect(wrapper.state().tagFilter).toEqual(null);
  });

  it('should set a date filter', () => {
    wrapper.instance().setDateFilter('12/25/2018');
    expect(wrapper.state().dateFilter).toEqual('12/25/2018');
  });

  it('should set the dateFilter to null if the value is all', () => {
    wrapper.instance().setDateFilter('all');
    expect(wrapper.state().dateFilter).toEqual(null);
  });
});
