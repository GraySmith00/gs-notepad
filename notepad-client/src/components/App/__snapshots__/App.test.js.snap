// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          NotePad
        </h1>,
        <NoteForm
          addNote={[Function]}
        />,
        <DateFilter
          notes={
            Array [
              Object {
                "createdAt": 1545951483012,
                "id": "dd5aa440-0a2a-11e9-93fa-a7949507e226",
                "tag": "hobby",
                "text": "buy groceries",
              },
              Object {
                "createdAt": 1545951433152,
                "id": "bfa29c00-0a2a-11e9-93fa-a7949507e226",
                "tag": "work",
                "text": "do laundry",
              },
              Object {
                "createdAt": 1545951460092,
                "id": "cfb153c0-0a2a-11e9-93fa-a7949507e226",
                "tag": "hobby",
                "text": "play golf",
              },
              Object {
                "createdAt": 1545951504338,
                "id": "ea10bb20-0a2a-11e9-93fa-a7949507e226",
                "tag": "personal",
                "text": "buy gifts",
              },
              Object {
                "createdAt": 1545451632475,
                "id": "425ca4b0-0a2b-11e9-93fa-a7949507e226",
                "tag": "personal",
                "text": "set up tv",
              },
              Object {
                "createdAt": 1545951440518,
                "id": "c4069260-0a2a-11e9-93fa-a7949507e226",
                "tag": "work",
                "text": "run errands",
              },
            ]
          }
          setDateFilter={[Function]}
        />,
        <TagFilter
          setTagFilter={[Function]}
        />,
        <NotesList
          dateFilter={null}
          notes={
            Array [
              Object {
                "createdAt": 1545951483012,
                "id": "dd5aa440-0a2a-11e9-93fa-a7949507e226",
                "tag": "hobby",
                "text": "buy groceries",
              },
              Object {
                "createdAt": 1545951433152,
                "id": "bfa29c00-0a2a-11e9-93fa-a7949507e226",
                "tag": "work",
                "text": "do laundry",
              },
              Object {
                "createdAt": 1545951460092,
                "id": "cfb153c0-0a2a-11e9-93fa-a7949507e226",
                "tag": "hobby",
                "text": "play golf",
              },
              Object {
                "createdAt": 1545951504338,
                "id": "ea10bb20-0a2a-11e9-93fa-a7949507e226",
                "tag": "personal",
                "text": "buy gifts",
              },
              Object {
                "createdAt": 1545451632475,
                "id": "425ca4b0-0a2b-11e9-93fa-a7949507e226",
                "tag": "personal",
                "text": "set up tv",
              },
              Object {
                "createdAt": 1545951440518,
                "id": "c4069260-0a2a-11e9-93fa-a7949507e226",
                "tag": "work",
                "text": "run errands",
              },
            ]
          }
          tagFilter={null}
        />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "NotePad",
        },
        "ref": null,
        "rendered": "NotePad",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "addNote": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "notes": Array [
            Object {
              "createdAt": 1545951483012,
              "id": "dd5aa440-0a2a-11e9-93fa-a7949507e226",
              "tag": "hobby",
              "text": "buy groceries",
            },
            Object {
              "createdAt": 1545951433152,
              "id": "bfa29c00-0a2a-11e9-93fa-a7949507e226",
              "tag": "work",
              "text": "do laundry",
            },
            Object {
              "createdAt": 1545951460092,
              "id": "cfb153c0-0a2a-11e9-93fa-a7949507e226",
              "tag": "hobby",
              "text": "play golf",
            },
            Object {
              "createdAt": 1545951504338,
              "id": "ea10bb20-0a2a-11e9-93fa-a7949507e226",
              "tag": "personal",
              "text": "buy gifts",
            },
            Object {
              "createdAt": 1545451632475,
              "id": "425ca4b0-0a2b-11e9-93fa-a7949507e226",
              "tag": "personal",
              "text": "set up tv",
            },
            Object {
              "createdAt": 1545951440518,
              "id": "c4069260-0a2a-11e9-93fa-a7949507e226",
              "tag": "work",
              "text": "run errands",
            },
          ],
          "setDateFilter": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "setTagFilter": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "dateFilter": null,
          "notes": Array [
            Object {
              "createdAt": 1545951483012,
              "id": "dd5aa440-0a2a-11e9-93fa-a7949507e226",
              "tag": "hobby",
              "text": "buy groceries",
            },
            Object {
              "createdAt": 1545951433152,
              "id": "bfa29c00-0a2a-11e9-93fa-a7949507e226",
              "tag": "work",
              "text": "do laundry",
            },
            Object {
              "createdAt": 1545951460092,
              "id": "cfb153c0-0a2a-11e9-93fa-a7949507e226",
              "tag": "hobby",
              "text": "play golf",
            },
            Object {
              "createdAt": 1545951504338,
              "id": "ea10bb20-0a2a-11e9-93fa-a7949507e226",
              "tag": "personal",
              "text": "buy gifts",
            },
            Object {
              "createdAt": 1545451632475,
              "id": "425ca4b0-0a2b-11e9-93fa-a7949507e226",
              "tag": "personal",
              "text": "set up tv",
            },
            Object {
              "createdAt": 1545951440518,
              "id": "c4069260-0a2a-11e9-93fa-a7949507e226",
              "tag": "work",
              "text": "run errands",
            },
          ],
          "tagFilter": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            NotePad
          </h1>,
          <NoteForm
            addNote={[Function]}
          />,
          <DateFilter
            notes={
              Array [
                Object {
                  "createdAt": 1545951483012,
                  "id": "dd5aa440-0a2a-11e9-93fa-a7949507e226",
                  "tag": "hobby",
                  "text": "buy groceries",
                },
                Object {
                  "createdAt": 1545951433152,
                  "id": "bfa29c00-0a2a-11e9-93fa-a7949507e226",
                  "tag": "work",
                  "text": "do laundry",
                },
                Object {
                  "createdAt": 1545951460092,
                  "id": "cfb153c0-0a2a-11e9-93fa-a7949507e226",
                  "tag": "hobby",
                  "text": "play golf",
                },
                Object {
                  "createdAt": 1545951504338,
                  "id": "ea10bb20-0a2a-11e9-93fa-a7949507e226",
                  "tag": "personal",
                  "text": "buy gifts",
                },
                Object {
                  "createdAt": 1545451632475,
                  "id": "425ca4b0-0a2b-11e9-93fa-a7949507e226",
                  "tag": "personal",
                  "text": "set up tv",
                },
                Object {
                  "createdAt": 1545951440518,
                  "id": "c4069260-0a2a-11e9-93fa-a7949507e226",
                  "tag": "work",
                  "text": "run errands",
                },
              ]
            }
            setDateFilter={[Function]}
          />,
          <TagFilter
            setTagFilter={[Function]}
          />,
          <NotesList
            dateFilter={null}
            notes={
              Array [
                Object {
                  "createdAt": 1545951483012,
                  "id": "dd5aa440-0a2a-11e9-93fa-a7949507e226",
                  "tag": "hobby",
                  "text": "buy groceries",
                },
                Object {
                  "createdAt": 1545951433152,
                  "id": "bfa29c00-0a2a-11e9-93fa-a7949507e226",
                  "tag": "work",
                  "text": "do laundry",
                },
                Object {
                  "createdAt": 1545951460092,
                  "id": "cfb153c0-0a2a-11e9-93fa-a7949507e226",
                  "tag": "hobby",
                  "text": "play golf",
                },
                Object {
                  "createdAt": 1545951504338,
                  "id": "ea10bb20-0a2a-11e9-93fa-a7949507e226",
                  "tag": "personal",
                  "text": "buy gifts",
                },
                Object {
                  "createdAt": 1545451632475,
                  "id": "425ca4b0-0a2b-11e9-93fa-a7949507e226",
                  "tag": "personal",
                  "text": "set up tv",
                },
                Object {
                  "createdAt": 1545951440518,
                  "id": "c4069260-0a2a-11e9-93fa-a7949507e226",
                  "tag": "work",
                  "text": "run errands",
                },
              ]
            }
            tagFilter={null}
          />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "NotePad",
          },
          "ref": null,
          "rendered": "NotePad",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "addNote": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "notes": Array [
              Object {
                "createdAt": 1545951483012,
                "id": "dd5aa440-0a2a-11e9-93fa-a7949507e226",
                "tag": "hobby",
                "text": "buy groceries",
              },
              Object {
                "createdAt": 1545951433152,
                "id": "bfa29c00-0a2a-11e9-93fa-a7949507e226",
                "tag": "work",
                "text": "do laundry",
              },
              Object {
                "createdAt": 1545951460092,
                "id": "cfb153c0-0a2a-11e9-93fa-a7949507e226",
                "tag": "hobby",
                "text": "play golf",
              },
              Object {
                "createdAt": 1545951504338,
                "id": "ea10bb20-0a2a-11e9-93fa-a7949507e226",
                "tag": "personal",
                "text": "buy gifts",
              },
              Object {
                "createdAt": 1545451632475,
                "id": "425ca4b0-0a2b-11e9-93fa-a7949507e226",
                "tag": "personal",
                "text": "set up tv",
              },
              Object {
                "createdAt": 1545951440518,
                "id": "c4069260-0a2a-11e9-93fa-a7949507e226",
                "tag": "work",
                "text": "run errands",
              },
            ],
            "setDateFilter": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "setTagFilter": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "dateFilter": null,
            "notes": Array [
              Object {
                "createdAt": 1545951483012,
                "id": "dd5aa440-0a2a-11e9-93fa-a7949507e226",
                "tag": "hobby",
                "text": "buy groceries",
              },
              Object {
                "createdAt": 1545951433152,
                "id": "bfa29c00-0a2a-11e9-93fa-a7949507e226",
                "tag": "work",
                "text": "do laundry",
              },
              Object {
                "createdAt": 1545951460092,
                "id": "cfb153c0-0a2a-11e9-93fa-a7949507e226",
                "tag": "hobby",
                "text": "play golf",
              },
              Object {
                "createdAt": 1545951504338,
                "id": "ea10bb20-0a2a-11e9-93fa-a7949507e226",
                "tag": "personal",
                "text": "buy gifts",
              },
              Object {
                "createdAt": 1545451632475,
                "id": "425ca4b0-0a2b-11e9-93fa-a7949507e226",
                "tag": "personal",
                "text": "set up tv",
              },
              Object {
                "createdAt": 1545951440518,
                "id": "c4069260-0a2a-11e9-93fa-a7949507e226",
                "tag": "work",
                "text": "run errands",
              },
            ],
            "tagFilter": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
